src/config.c

```diff
@@ -907,13 +907,20 @@ bool config_find_or_browse_install_directory()
 		gConfigGeneral.game_path = malloc(strlen(path) + 1);
 		safe_strcpy(gConfigGeneral.game_path, path, MAX_PATH);
 	} else {
-		platform_show_messagebox("Unable to find RCT2 installation directory. Please select the directory where you installed RCT2!");
-		installPath = platform_open_directory_browser("Please select your RCT2 directory");
-		if (installPath == NULL)
-			return false;
-
-		SafeFree(gConfigGeneral.game_path);
-		gConfigGeneral.game_path = installPath;
+		while (1) {
+			platform_show_messagebox("Unable to find RCT2 installation directory. Please select the directory where you installed RCT2!");
+			installPath = platform_open_directory_browser("Please select your RCT2 directory");
+			if (installPath == NULL)
+				return false;
+
+			SafeFree(gConfigGeneral.game_path);
+			gConfigGeneral.game_path = installPath;
+			
+			if (platform_original_game_data_exists(installPath))
+				return true;
+
+			platform_show_messagebox("Could not find Data\\g1.dat at this address");
```

Comment:
I think it would be nice to incorporate user's selection into the message (`Cannot find /xxx/yyy/zzz/Data/g1.dat` instead of `Data/g1.dat`). Also: `platform_get_path_separator()`.